Lib/test/test_perf_profiler.py

```diff
@@ -0,0 +1,348 @@
+import unittest
+import subprocess
+import sys
+import sysconfig
+import os
+import pathlib
+from test import support
+from test.support.script_helper import (
+    make_script,
+    assert_python_failure,
+    assert_python_ok,
+)
+from test.support.os_helper import temp_dir
+
+
+if not support.has_subprocess_support:
+    raise unittest.SkipTest("test module requires subprocess")
+
+
+def supports_trampoline_profiling():
+    perf_trampoline = sysconfig.get_config_var("PY_HAVE_PERF_TRAMPOLINE")
+    if not perf_trampoline:
+        return False
+    return int(perf_trampoline) == 1
+
+
+if not supports_trampoline_profiling():
+    raise unittest.SkipTest("perf trampoline profiling not supported")
+
+
+class TestPerfTrampoline(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.perf_files = set(pathlib.Path("/tmp/").glob("perf-*.map"))
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        files_to_delete = (
+            set(pathlib.Path("/tmp/").glob("perf-*.map")) - self.perf_files
+        )
+        for file in files_to_delete:
+            file.unlink()
+
+    def test_trampoline_works(self):
+        code = """if 1:
+                def foo():
+                    pass
+
+                def bar():
+                    foo()
+
+                def baz():
+                    bar()
+
+                baz()
+                """
+        with temp_dir() as script_dir:
+            script = make_script(script_dir, "perftest", code)
+            with subprocess.Popen(
```

Comment:
Use `script_helpers.run_python_until_end('-Xperf', script)` in these rather than `subprocess.Popen` in these tests - it takes care of common things to allow a child interpreter process to work in various environments. https://github.com/python/cpython/blob/main/Lib/test/support/script_helper.py#L94

Reply:
Unfortunately we need the PID (see following code) and `script_helpers.run_python_until_end('-Xperf', script)` returns a namedtuple of `(rc, stdout, stderr)`. I am missing something?

Reply:
oh right, the pid!  I guess leave these then.  A new helper could be added but until we see this test fall over with problems somewhere because of that, no big deal then.  (text=True the calls at least)